services/fhir.ts

```diff
@@ -225,35 +225,39 @@ export function makeReference<T extends AidboxResource>(
 export function isReference<T extends AidboxResource>(
     resource: T | AidboxReference<T>
 ): resource is AidboxReference<T> {
-    return _.isEmpty(
-        _.difference(_.keys(resource), [
-            'id',
-            'resourceType',
-            '_id',
-            'resource',
-            'display',
-            'identifier',
-            'uri',
-            'localRef',
-            'extension',
-        ])
-    );
+    return !Object.keys(resource).filter(
+        (resource) =>
+            ['id', 'resourceType', '_id', 'resource', 'display', 'identifier', 'uri', 'localRef', 'extension'].indexOf(
+                resource
+            ) === -1
+    ).length;
 }
 
 export type ResourcesMap<T extends AidboxResource> = { [x: string]: T[] | undefined };
 
 export function extractBundleResources<T extends AidboxResource>(bundle: Bundle<T>): ResourcesMap<T> {
-    const entriesByResourceType = _.groupBy(bundle.entry, (entry) => entry.resource!.resourceType);
+    const entriesByResourceType = {};
 
-    return _.mapValues(entriesByResourceType, (entries) => _.map(entries, (entry) => entry.resource!));
+    bundle.entry?.forEach(function(entry) {
```

Comment:
bundle.entry? -- what does it mean?

Reply:
By the way, bundle.entry can be undefined following the spec. Let's check it before.